circle.yml

```diff
@@ -16,8 +16,8 @@ dependencies:
     - ~/.cache/yarn
   override:
     - yarn install --no-progress
-    - cd example && yarn install --no-progress
+#    - cd example && yarn install --no-progress
 test:
   override:
     - yarn run test:ci
-    - cd example && yarn run test:ci && yarn run test:coverage
+#    - cd example && yarn run test:ci && yarn run test:coverage
```

Comment:
example tests need to run, please uncomment

Reply:
As discussed in [#204](https://github.com/thymikee/jest-preset-angular/pull/201#issuecomment-442205239) and [described earlier](https://github.com/thymikee/jest-preset-angular/pull/204#issuecomment-442272185), the tests will not run until this code itself is published as `jest-preset-angular`, as it is self referencing but going through a breaking change right now.

I can only make them run using
```
cd example && \
npm install && \
npm install wtho/jest-preset-angular#ts-jest-23-10 && \
yarn run test:coverage
```
or
```
cd example && \
yarn install && \
yarn remove jest-preset-angular#ts-jest-23-10 && \
yarn add wtho/jest-preset-angular#ts-jest-23-10 && \
cp ../InlineHtmlStripStylesTransformer.js node_modules/jest-preset-angular && \
yarn run test:coverage
```
as the example-app references the version on npm, which is in version 6, but this is a breaking change.

Maybe you have another idea?

Reply:
We used to link the library, but it failed in some cases if I remember well. Should revisit that, either by using yarn link or installing the preset from directory instead of npm package

Reply:
Linking alone did not work for me, some packages were still missing, but I found a workaround.

### Did not work
On linking or setting `"preset": "/my/absolut/path/of/jest-preset-angular"` in package.json:
```
    Cannot find module 'zone.js/dist/zone.js' from 'index.js'

      2 | require('core-js/es6/reflect');
      3 | require('core-js/es7/reflect');
    > 4 | require('zone.js/dist/zone.js');
        | ^
```
After adding zone.js to the root-project it complains about `@angular/core`, then `rxjs`, `@angular/platform-browser-dynamic`, `@angular/platform-browser`, `@angular/common`, `@angular/compiler`, ...

On `"preset": ".."` in package.json:
```
● Validation Error:
  Module jest-preset-angular should have "jest-preset.js" or "jest-preset.json" file at the root.
```

### Worked
From the root of this repo, after both example app and root had run their initial installs:
```
npm install example/node_modules/@angular/common example/node_modules/@angular/compiler example/node_modules/@angular/core example/node_modules/@angular/platform-browser example/node_modules/rxjs example/node_modules/zone.js

npm link
cd example
npm link jest-preset-angular
npm test
```
I like this approach as it makes sure it uses the versions of the example app when running the tests and we do not pollute the root package.json (ok, when running tests we do, but generally not)

If you are ok with this, I will update the CI configuration and test it on the circleci.

Reply:
Just merged it, feel free to rebase and get over this issue :)

Reply:
Rebased, and works, awesome!